tap_pardot/streams.py

```diff
@@ -376,6 +376,24 @@ class Prospects(UpdatedAtReplicationStream):
 
     is_dynamic = False
 
+    def get_records(self):
+        offset = 0
+        params = self.get_params()
+        while True:
+            params["offset"] = offset
+            data = self.client.get(self.endpoint, **params)
+            records = data["result"]
+            if not records:
+                break
+            offset += 200
+            yield from records[self.data_key]
+
+    def sync(self):
+        self.pre_sync()
+        for rec in self.sync_page():
+            yield rec
+        self.post_sync()
```

Comment:
Since the `pre_sync` and `post_sync` functions are not overridden in the parent class, they seem unnecessary here. Would it be beneficial to remove them from the class definition?
Correct me here if I am wrong.

Reply:
pre_sync and post_sync are not implemented/modified in this class, but do exist in the parent class, and can be modified to add/remove functionality.
i am not removing it to keep this pattern constant across all streams of the tap.

Reply:
@sgandhi1311 please re-review